### PR TITLE
Bring back HNSWLIB_NO_NATIVE

### PR DESCRIPTION
**Pull request by dyashuni**
Brining back `HNSWLIB_NO_NATIVE` to fix issue https://github.com/nmslib/hnswlib/pull/476

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -73,15 +73,19 @@ def cpp_flag(compiler):
 
 class BuildExt(build_ext):
     """A custom build extension for adding compiler-specific options."""
+    native_flag = '-march=native'
```
Comment (yurymalkov) [inline setup.py:76]: Can you please name it `compiler_flag_native`?

File: setup.py [hunk 2]
```diff
@@ -97,23 +101,23 @@ def build_extensions(self):
...
-                    if has_flag(self.compiler, m1_flag):
-                        print('adding flag:', m1_flag)
-                        opts.append(m1_flag)
-                    else:
-                        print(f'flag: {m1_flag} is not available')
-            else:
-                print(f'flag: {native_flag} is available')
+            if not os.environ.get("HNSWLIB_NO_NATIVE"):
+                # check that native flag is available
+                print('checking avalability of flag:', self.native_flag)
```
Comment (yurymalkov) [inline setup.py:106]: I think it `BuildExt.native_flag` is better than `self.native_flag` for readability (as `self.native_flag` is never set)

File: setup.py (hunk 1 above)
Comment (dyashuni) [reply setup.py]: Done

**Comment by xhochy** [comment]
@dyashuni Thank you for picking this up. I just came back from leave and it's nice to see someone took this.

**Comment by yurymalkov** [comment]
Thank you!

**Pull request merged**